components/script/dom/mediaquerylist.rs

```diff
@@ -138,16 +138,24 @@ impl WeakMediaQueryListVec {
     /// Evaluate media query lists and report changes
     /// https://drafts.csswg.org/cssom-view/#evaluate-media-queries-and-report-changes
     pub fn evaluate_and_report_changes(&self) {
-        for mql in self.cell.borrow().iter() {
+        rooted_vec!(let mut mql_list);
+        for mql in self.cell.borrow_mut().iter() {
             let mql = mql.root().unwrap();
             if let MediaQueryListMatchState::Changed(_) = mql.evaluate_changes() {
-                let event = MediaQueryListEvent::new(&mql.global(),
-                                                     atom!("change"),
-                                                     false, false,
-                                                     mql.Media(),
-                                                     mql.Matches());
-                event.upcast::<Event>().fire(mql.upcast::<EventTarget>());
+                // Recording list of changed Media Queries
+                mql_list.push(JS::from_ref(&*mql));
             }
+
+        }
+        // Sending change events for all changed Media Queries
+        for mql in mql_list.iter()
+        {
```

Comment:
@jdm For the sake of theory, there is no performance difference between the two right?